client/components/editor/teaching-elements/Video.vue

```diff
@@ -35,74 +40,70 @@
 <script>
 import { extname } from 'path';
 import get from 'lodash/get';
-import 'videojs-youtube';
-import 'videojs-vimeo';
-import { videoPlayer as VideoPlayer } from 'vue-video-player';
+import { Plyr } from 'vue-plyr';
+
 const MediaError = window.MediaError;
 
+const NOT_NATIVE = /youtu\.?be|vimeo/;
+const CUSTOM_TYPE_MAPPING = {
+  ogv: 'ogg'
+};
+
 export default {
   name: 'te-video',
-  props: ['element', 'isFocused', 'isDragged'],
+  props: {
+    element: { type: Object, required: true },
+    isFocused: { type: Boolean, default: false },
+    isDragged: { type: Boolean, default: false }
+  },
   data() {
-    return { error: null };
+    return {
+      error: null,
+      switchingVideo: false
+    };
   },
   computed: {
     player() {
-      return this.$refs.video && this.$refs.video.player;
-    },
-    source() {
-      const src = get(this.element, 'data.url');
-      if (!src) return;
-      const type = getMimetype(src);
-      return { type, src };
+      return get(this.$refs, 'video.player');
     },
-    poster() {
-      return get(this.element, 'data.poster');
+    url() {
+      return get(this.element, 'data.url', '');
     },
-    options() {
-      return {
-        autoplay: false,
-        techOrder: ['html5', 'youtube', 'vimeo'],
-        poster: this.poster,
-        sources: [this.source]
-      };
+    type() {
+      if (NOT_NATIVE.test(this.url)) return { isNative: false };
+      const ext = extname(this.url).substring(1);
+      const name = `video/${CUSTOM_TYPE_MAPPING[ext] || ext}`;
```

Comment:
This is actually [`mimetype=type/subtype`](https://developer.mozilla.org/en-US/docs/Web/HTTP/Basics_of_HTTP/MIME_types#General_structure) so name those variables accordingly :bowtie: